service/src/priority.rs

```diff
@@ -48,7 +48,9 @@ impl PartialEq for Priority {
 impl Eq for Priority {}
 
 impl PartialOrd for Priority {
+    #[allow(clippy::non_canonical_partial_ord_impl)]
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        // todo: what is going on here ?
```

Comment:
Is there something specific that is confusing here about the comparison? If you are asking about the second element in the tuple, it is adjusting the comparison of the match scores when an element is used more recently or more often.

Reply:
I just added it because i added `#[allow(clippy::non_canonical_partial_ord_impl)]` to remove the warning without understanding the code. Btw, this code is part of this pr #219